app/uk/gov/hmrc/fileupload/controllers/EnvelopeController.scala

```diff
@@ -27,27 +29,41 @@ import uk.gov.hmrc.fileupload.utils.JsonUtils.jsonBodyParser
 import uk.gov.hmrc.fileupload.write.envelope._
 import uk.gov.hmrc.fileupload.write.infrastructure._
 import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, _}
-import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
 
+
+package object envelope {
+
+  import play.api.libs.concurrent.Execution.Implicits._
+
+  object EnvelopeController extends EnvelopeController(
+    withBasicAuth = MicroserviceGlobal.withBasicAuth,
```

Comment:
I do understand that the MicroserviceGlobal needs a bit of cleaning up, but I think we should not introduce a dependency to MicroserviceGlobal within our controllers. One reason is the implict execution context I mentioned in another comment. I think the controller should not know where the dependencies come from and how they are created (ie. if they rely on being lazy or not). Some dependencies are initalised after startup (ie withBasicAuth, when the configuration comes available). What do you guys think?